src/ExtensionSet.php

```diff
@@ -142,7 +143,11 @@ public function getLastModified(): int
 
     public function addExtension(ExtensionInterface $extension): void
     {
-        $class = $extension::class;
+        if ($extension instanceof AttributeExtension) {
+            $class = $extension->getClass();
+        } else {
+            $class = $extension::class;
+        }
```

Comment:
This is for cache invalidation when an extension is added/removed.

Instead of relying on this class check, I can introduce an interface: `DelegatingExtensionInterface` with the `getClass` method.